arch/aarch64/plthook.S

```diff
@@ -76,16 +76,29 @@ END(plt_hooker)
 ENTRY(plthook_return)
 	/* setup frame pointer */
 	stp	x29, x30, [sp, #-16]!
-	stp	x0, x8, [sp, #-16]!
-	str	d0, [sp, #-16]!
 
-	add	x0, sp, #16
+	/* save return values */
+	stp	x0, x1, [sp, #-16]!
+	stp	d0, d1, [sp, #-16]!
+
+	/*
+	 * save indirect result location register
+	 * used in C++ for returning non-trivial objects
+	 */
+	str	x8, [sp, #-16]!
+
+	add	x0, sp, #32
```

Comment:
@namhyung Actually, I don't clearly understand this calculation. Could you please confirm this?

Reply:
`x0` is the first argument of `plthook_exit` (`long *retval`) so it should point the location where original `x0` is saved which is `sp + 16` in this case.  But I'm not sure it should save `x8` register as well as it's the "indirect result location register".

Reply:
Oh, I missed this commit.  It look ok now.

Reply:
> But I'm not sure it should save x8 register as well as it's the "indirect result location register".

I'm not 100% sure as well, but I think the same problem can be happened because when library calls are done indirectly, but their return is made directly. So the library function will immediately return its return value back to the caller.

Reply:
The node test works fine even without saving `x8` in `plthook.S` but just added to prevent any kind of future problem.

Reply:
If you're not sure about this, I can drop this as of now because it's more important to fix the current problem first.

Reply:
I think we should take the safer way and the original commit already did it.

Reply:
So do you mean to keep the second commit for `plthook.S` as is?

Reply:
Yep

Reply:
Thanks. Have a good night!